salt/metalk8s/internal/sreport/files/metalk8s.py

```diff
@@ -0,0 +1,123 @@
+                self.add_cmd_output('%s %s' % (kube_cmd, subcmd))
+
+            # get all namespaces in use
+            kn = self.get_command_output('%s get namespaces' % kube_cmd)
+            knsps = [n.split()[0] for n in kn['output'].splitlines()[1:] if n]
+
+            resources = [
+                'pods',
+                'deploy',
+                'rc',
+                'services',
+                'ds',
+                'cm',
+            ]
+
+            # nodes and pvs are not namespaced, must pull separately.
+            # Also collect master metrics
+            self.add_cmd_output([
+                "{} get -o json nodes".format(kube_cmd),
+                "{} get -o json pv".format(kube_cmd),
+                "{} get --raw /metrics".format(kube_cmd)
+            ])
+
+            for n in knsps:
+                knsp = '--namespace=%s' % n
+                if self.get_option('all'):
+                    k_cmd = '%s %s %s' % (kube_cmd, kube_get_cmd, knsp)
+
+                    self.add_cmd_output('%s events' % k_cmd)
+
+                    for res in resources:
+                        self.add_cmd_output('%s %s' % (k_cmd, res))
+
+                    if self.get_option('describe'):
+                        # need to drop json formatting for this
+                        k_cmd = '%s get %s' % (kube_cmd, knsp)
+                        for res in resources:
+                            r = self.get_command_output(
+                                '%s %s' % (k_cmd, res))
+                            if r['status'] == 0:
+                                k_list = [k.split()[0] for k in
+                                          r['output'].splitlines()[1:]]
+                                for k in k_list:
+                                    k_cmd = '%s %s' % (kube_cmd, knsp)
+                                    self.add_cmd_output(
+                                        '%s describe %s %s' % (k_cmd, res, k))
+
+                if self.get_option('podlogs'):
+                    k_cmd = '%s %s' % (kube_cmd, knsp)
+                    r = self.get_command_output('%s get pods' % k_cmd)
+                    if r['status'] == 0:
+                        pods = [p.split()[0] for p in
+                                r['output'].splitlines()[1:]]
+                        for pod in pods:
+                            self.add_cmd_output('%s logs %s' % (k_cmd, pod))
+
+            if not self.get_option('all'):
+                k_cmd = '%s get --all-namespaces=true' % kube_cmd
+                for res in resources:
+                    self.add_cmd_output('%s %s' % (k_cmd, res))
+
+    def postproc(self):
+        # First, clear sensitive data from the json output collected.
+        # This will mask values when the "name" looks susceptible of
+        # values worth obfuscating, i.e. if the name contains strings
+        # like "pass", "pwd", "key" or "token"
+        env_regexp = r'(?P<var>{\s*"name":\s*[^,]*' \
```

Comment:
Not very readable/understandable.

Either give an example string of what it's supposed to match, or use the verbose regexp mode that allows to insert comments in the regexp.